src/libsystemd/sd-device/test-sd-device.c

```diff
@@ -499,6 +499,37 @@ TEST(sd_device_enumerator_add_match_parent) {
         }
 }
 
+TEST(sd_device_enumerator_add_all_parents) {
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        int r;
+
+        assert_se(sd_device_enumerator_new(&e) >= 0);
+        assert_se(sd_device_enumerator_allow_uninitialized(e) >= 0);
+        /* See comments in TEST(sd_device_enumerator_devices). */
+        assert_se(sd_device_enumerator_add_match_subsystem(e, "bdi", false) >= 0);
+        assert_se(sd_device_enumerator_add_nomatch_sysname(e, "loop*") >= 0);
+        assert_se(sd_device_enumerator_add_match_subsystem(e, "net", false) >= 0);
+
+        /*enable all_parents*/
+        assert_se(sd_device_enumerator_add_all_parents(e) >= 0);
```

Comment:
I'd like to make first enumerate devices without this, and confirm actually only block devices are enumerated.
Then, later enable this and confirm that the parent devices that does not match the conditions will enumerated.